rfcs/2022-04-09-accessibility.md

```diff
@@ -0,0 +1,495 @@
+
+When set to `0`, the element becomes focusable by keyboard and via programmatic means with `focus()`. 
+When set to `-1`, the element becomes focusable programmatically, but it does not become part of the keyboard focus order.
+
+The following table describes tabindex behavior in modern browsers:
+
+| tabindex attribute |	Focusable with mouse or programmatically |	Tab navigable |
+| :-- | :-- | :-- |
+| not present	| The user agent will decide | The user agent will decide |
+| Negative (tabindex="-1")	| Yes | No, can only be focused programmatically |
+| Zero (tabindex="0") | Yes | Yes, In tab order relative to element's position in document | 
+| Positive (tabindex="2") | Yes | Yes, tabindex value determines where this element is positioned in the tab order | 
+
+> Warning: Avoid using positive values for tabindex. Using positive values means authors will have to set (and maintain) tabindex values for all focusable elements on the page whenever they use one or more positive values for tabindex.  
+
+> *[Failure of Success Criterion 2.4.3](https://www.w3.org/WAI/WCAG21/Techniques/failures/F44) due to using tabindex to create a tab order that does not preserve meaning and operability*
+
+
+## Skip navigations
+
+> [Specification](https://www.w3.org/TR/2013/NOTE-WCAG20-TECHS-20130905/G1)
+
+A skip navigation link provides a way for keyboard and screen reader users to skip to main content of a webpage.  
+Without skip links, keyboard and screen reader users must navigate a long list of navigation links and other elements before ever arriving at the main content. This can be particularly difficult for users with some forms of motor disabilities.
+
+**Temporarily hidden skip links**
+
+Most common type of skip links are the ones that are hidden until the user navigates to it with a keyboard.
+
+The link must be:
+
+- Hidden by default
+- Accessible to keyboard navigation
+- Visible when it is focused
+- Set focus to the main content area when activated
+
+### Implementations
+
+Skip nav implementations are fairly simple: 
+
+> [Live Demo](https://stackblitz.com/edit/react-ts-nnnymu?file=App.tsx)
+
+```html
+<body>
+  <a class='skip-link' href='#main-content'>
+    Skip Navigation or Skip to Content
+  </a>
+  <main id='main-content'>
+    Content here
+  </main>
+</body>
+```
+
+The `skip-link` class here is a screen reader only [visually hidden](https://webaim.org/techniques/css/invisiblecontent/) class, which is only accessible through screen readers. By using the `:focus` pesudo class it becomes visible when focused.
+
+- [ReachUI SkipNav](https://reach.tech/skip-nav/)
+- [ChakraUI SkipNav](https://chakra-ui.com/docs/components/navigation/skip-nav)
+
+## Composite widgets
+
+A composite is a widget that may contain navigable descendants or childrens. Composite widgets should only have single navigation tab stop.
```

Comment:
too much content but difficult to visualise as to what you mean here. probably a small gif or video would help understand this clearly

Reply:
Sure.